File: Top_K_Frequent_Elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        dt = {}
        for e in nums:
            if e not in dt:
                dt[e] = 0
            dt[e] += 1
        ls = sorted(dt.items(), key=lambda x: -x[1])
        return [x[0] for x in ls[:k]]

File: java_version/src/TopKFrequentElements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<Integer, Integer>();
        for(int x: nums){
            count.put(x, count.getOrDefault(x, 0) + 1);
        }
        
        Queue<Integer> minHeap = new PriorityQueue<Integer>((a, b) -> Integer.compare(count.get(a), count.get(b)));
        for (int x: count.keySet()){
            if (minHeap.size() == k){
                if (count.get(minHeap.peek()) < count.get(x)){
                    minHeap.remove();                    
                    minHeap.add(x);
                }
            }
            else{
                minHeap.add(x);
            }
        }
        int[] res = new int[k];
        for(int i=0; i< k; i++){
            res[i] = minHeap.remove();
        }
        return res;
    }
}